app/uk/gov/hmrc/economiccrimelevyregistration/navigation/contacts/SecondContactNamePageNavigator.scala

```diff
@@ -18,17 +18,25 @@ package uk.gov.hmrc.economiccrimelevyregistration.navigation.contacts
 
 import play.api.mvc.Call
 import uk.gov.hmrc.economiccrimelevyregistration.controllers.{contacts, routes}
-import uk.gov.hmrc.economiccrimelevyregistration.models.Registration
+import uk.gov.hmrc.economiccrimelevyregistration.models.{CheckMode, NormalMode, Registration}
 import uk.gov.hmrc.economiccrimelevyregistration.navigation.PageNavigator
 
 class SecondContactNamePageNavigator extends PageNavigator {
 
   override protected def navigateInNormalMode(registration: Registration): Call =
     registration.contacts.secondContactDetails.name match {
-      case Some(_) => contacts.routes.SecondContactRoleController.onPageLoad()
+      case Some(_) => contacts.routes.SecondContactRoleController.onPageLoad(NormalMode)
       case _       => routes.JourneyRecoveryController.onPageLoad()
     }
 
-  override protected def navigateInCheckMode(registration: Registration): Call = ???
+  override protected def navigateInCheckMode(registration: Registration): Call =
+    registration.contacts.secondContactDetails.name match {
+      case Some(_) =>
+        registration.contacts.secondContactDetails.role match {
```

Comment:
similar to my other comment I think we need to be matching on the other 3 contact details here